### DISCUSSION TITLE
How to Mock mapperly for unit testing?

### DISCUSSION THREAD

**Discussion by ramarahul** [riok/mapperly#1423] [Q&A]
I have created a mapper class in a separate file as mentioned [here](https://mapperly.riok.app/docs/getting-started/first-mapper/). And I've used it in another class for mapping. However, when I'm testing my other class using Moq, I am getting an error saying that it is Non-Overridable even if I Mock mapperly too. What's the best way to unit-test mapperly?

**Comment by sveinhelge** [comment]
One of the main reasons I use Mapperly is the fact that I do not need to write unit test for the mappings. Mapperly does evaluate every property to ensure I have not forgotten any properties. When adding new property to source or target or renaming a property causing a mapping to be missed, I will get an error/warning (depending on my setting). If I have any special mapping used by Mapperly, I do unit tests those methods. But that's outside Mapperly.
But that said. You may have scenarios where unit testing of Mapplery make sense.

**Comment by latonz** [comment]
What do you mean with "even if I mock Mapperly too"?
If you want to unit-test Mapperly generated code you could write unit tests the same way you do it for other methods, just test/call the declared mapping methods. No need for mocking anything.
